### PR TITLE
[castai-cluster-controller] Add RBAC permissions to cluster-controller for castai-pod-pinner upgrades

### PR DESCRIPTION
**Pull request by kiriloman**

### CODE REVIEW

File: charts/castai-cluster-controller/templates/rbac.yaml [hunk 1]
```diff
@@ -73,11 +73,11 @@ rules:
   # For installing/updating CAST AI components.
   - apiGroups: [ "rbac.authorization.k8s.io" ]
     resources: [ "roles" ]
-    resourceNames: [ "castai-cluster-controller", "castai-evictor", "castai-agent", "castai-spot-handler", "castai-kvisor", "castai-egressd" ]
+    resourceNames: [ "castai-cluster-controller", "castai-evictor", "castai-agent", "castai-spot-handler", "castai-kvisor", "castai-egressd", "castai-pod-pinner" ]
```
Comment (aldor007) [inline charts/castai-cluster-controller/templates/rbac.yaml:76]: you also need to extend this list https://github.com/castai/cluster-controller/blob/613c33dab377d57c1e0f4acbb42f25bfe1bcd455/helm/hook/hook.go#L36

Comment (kiriloman) [reply charts/castai-cluster-controller/templates/rbac.yaml:76]: Updated: https://github.com/castai/cluster-controller/pull/89

We will also need to patch the `castai-pod-pinner` webhook, so I suppose we would need to add the below, correct?

```
  - apiGroups: [ "admissionregistration.k8s.io" ] #idk
    resourceNames: [ "castai-pod-pinner" ]
    resources: [ "mutatingwebhookconfigurations" ]
    verbs: [ "patch" ]
```

File: charts/castai-cluster-controller/Chart.yaml [hunk 2]
```diff
@@ -2,5 +2,5 @@ apiVersion: v2
 name: castai-cluster-controller
 description: Cluster controller is responsible for handling certain Kubernetes actions such as draining and deleting nodes, adding labels, approving CSR requests.
 type: application
-version: 0.55.2
+version: 0.55.3
 appVersion: "v0.39.3"
```
Comment (aldor007) [inline charts/castai-cluster-controller/Chart.yaml:6]: I would also update the cluster-controller version in this PR

File: charts/castai-cluster-controller/templates/rbac.yaml (hunk 1 above)
Comment (aldor007) [reply charts/castai-cluster-controller/templates/rbac.yaml:76]: Yep

**Review by aldor007 — APPROVED** [review]

**Pull request merged**